addons/sale_margin/models/sale_order.py

```diff
@@ -57,6 +57,7 @@ def create(self, vals):
 
     @api.depends('product_id', 'purchase_price', 'product_uom_qty', 'price_unit', 'price_subtotal')
     def _product_margin(self):
+        self.read(['price_subtotal', 'purchase_price', 'product_uom_qty', 'order_id'])
```

Comment:
please elaborate on that, I guess prefetching but it should be explicit in the commit. If you are going to make a read, why not actually using the result?